sdk/core/Azure.Core/src/Pipeline/HttpPipeline.cs

```diff
@@ -34,42 +33,47 @@ public HttpPipeline(HttpPipelineTransport transport, HttpPipelinePolicy[]? polic
         public Request CreateRequest()
             => _transport.CreateRequest();
 
+        public HttpPipelineMessage CreateMessage()
+        {
+            return new HttpPipelineMessage(CreateRequest(), ResponseClassifier);
+        }
+
         public ResponseClassifier ResponseClassifier { get; }
 
         public ClientDiagnostics Diagnostics { get; }
 
-        public ValueTask<Response> SendRequestAsync(Request request, CancellationToken cancellationToken)
+        public ValueTask SendAsync(HttpPipelineMessage message, CancellationToken cancellationToken)
         {
-            return SendRequestAsync(request, true, cancellationToken);
+            message.CancellationToken = cancellationToken;
+            return _pipeline.Span[0].ProcessAsync(message, _pipeline.Slice(1));
         }
 
-        public async ValueTask<Response> SendRequestAsync(Request request, bool bufferResponse, CancellationToken cancellationToken)
+        public void Send(HttpPipelineMessage message, CancellationToken cancellationToken)
         {
-            HttpPipelineMessage message = BuildMessage(request, bufferResponse, cancellationToken);
-            await _pipeline.Span[0].ProcessAsync(message, _pipeline.Slice(1)).ConfigureAwait(false);
-            return message.Response;
+            message.CancellationToken = cancellationToken;
+            _pipeline.Span[0].Process(message, _pipeline.Slice(1));
         }
 
-        public Response SendRequest(Request request, CancellationToken cancellationToken)
+        public async ValueTask<Response> SendRequestAsync(Request request, CancellationToken cancellationToken)
         {
-            return SendRequest(request, true, cancellationToken);
+            HttpPipelineMessage message = BuildMessage(request);
+            await SendAsync(message, cancellationToken).ConfigureAwait(false);
+            return message.Response;
         }
 
-        public Response SendRequest(Request request, bool bufferResponse, CancellationToken cancellationToken)
+        public Response SendRequest(Request request, CancellationToken cancellationToken)
         {
-            HttpPipelineMessage message = BuildMessage(request, bufferResponse, cancellationToken);
-            _pipeline.Span[0].Process(message, _pipeline.Slice(1));
+            HttpPipelineMessage message = BuildMessage(request);
+            Send(message, cancellationToken);
             return message.Response;
         }
 
-        private HttpPipelineMessage BuildMessage(Request request, bool bufferResponse, CancellationToken cancellationToken)
+        private HttpPipelineMessage BuildMessage(Request request)
         {
-            var message = new HttpPipelineMessage(request, ResponseClassifier, cancellationToken)
+            return new HttpPipelineMessage(request, ResponseClassifier)
             {
-                Request = request,
-                BufferResponse = bufferResponse
+                Request = request
```

Comment:
Does the `Request = request` add anything after passing it to the .ctor?

Reply:
Nope.